packages/enclave-contracts/tasks/ciphernode.ts

```diff
@@ -10,59 +10,496 @@ import { poseidon2 } from "poseidon-lite";
 
 export const ciphernodeAdd = task(
   "ciphernode:add",
-  "Register a ciphernode to the registry",
+  "Register a ciphernode to the bonding registry and ciphernode registry",
 )
   .addOption({
-    name: "ciphernodeAddress",
-    description: "address of ciphernode to register",
-    defaultValue: ZeroAddress,
+    name: "privateKey",
+    description:
+      "private key of the ciphernode to register (must have ENCL and USDC)",
+    defaultValue: "",
+  })
+  .addOption({
+    name: "licenseBondAmount",
+    description:
+      "amount of ENCL to bond (in wei, e.g., 1000000000000000000000 for 1000 ENCL)",
+    defaultValue: "1000000000000000000000",
+  })
+  .addOption({
+    name: "ticketAmount",
+    description:
+      "amount of USDC to deposit for tickets (in wei, e.g., 100000000 for 100 USDC)",
+    defaultValue: "1000000000",
   })
   .setAction(async () => ({
-    default: async ({ ciphernodeAddress }, hre) => {
-      const { deployAndSaveCiphernodeRegistryOwnable } = await import(
-        "../scripts/deployAndSave/ciphernodeRegistryOwnable"
+    default: async ({ privateKey, licenseBondAmount, ticketAmount }, hre) => {
+      const connection = await hre.network.connect();
+      const { ethers } = connection;
+
+      if (!privateKey) {
```

Comment:
why passing the key as an arg when we can use the first signer generated by hardhat with our private key in either a .env or better eventually using the keystore https://hardhat.org/docs/learn-more/configuration-variables#how-to-use-configuration-variables-and-the-keystore-plugin ?